From e44ff05b9c5cbadd0e36f0fe5e271cfd0c864406 Mon Sep 17 00:00:00 2001
From: Limeng <Meng.Li@windriver.com>
Date: Mon, 8 Jun 2020 15:54:12 +0800
Subject: [PATCH 3/9] driver: media: imx8: add valid_num_sensors variable to
 count the real number of sensors attached on board

Refer to nxp-imx-sdk5.4, commit 40682f78ec9a("media: staging:
imx: add media device driver support for IMX8 "), add valid_num_sensors
variable to count the real number of sensors attached on board.
Because the num_subdevs field of struct v4l2_async_notifier had been
removed, it is need to add another variable valid_num_sensors
to count the real number of camera sensors.

Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/media/platform/imx8/mxc-media-dev.c | 8 ++++----
 drivers/media/platform/imx8/mxc-media-dev.h | 1 +
 2 files changed, 5 insertions(+), 4 deletions(-)

diff --git a/drivers/media/platform/imx8/mxc-media-dev.c b/drivers/media/platform/imx8/mxc-media-dev.c
index 5562865330e7..0d83fc4eced7 100644
--- a/drivers/media/platform/imx8/mxc-media-dev.c
+++ b/drivers/media/platform/imx8/mxc-media-dev.c
@@ -336,10 +336,10 @@ static int subdev_notifier_bound(struct v4l2_async_notifier *notifier,
 
 	sensor->sd = sd;
 
-	mxc_md->num_sensors++;
+	mxc_md->valid_num_sensors++;
 
 	v4l2_info(&mxc_md->v4l2_dev, "Registered sensor subdevice: %s (%d)\n",
-		  sd->name, mxc_md->num_sensors);
+		  sd->name, mxc_md->valid_num_sensors);
 
 	return 0;
 }
@@ -873,7 +873,7 @@ static int mxc_md_probe(struct platform_device *pdev)
 		}
 
 		mxc_md->subdev_notifier.ops = &subdev_notifer_ops;
-		mxc_md->num_sensors = 0;
+		mxc_md->valid_num_sensors = 0;
 		mxc_md->link_status = 0;
 
 		ret = v4l2_async_notifier_register(&mxc_md->v4l2_dev,
@@ -884,7 +884,7 @@ static int mxc_md_probe(struct platform_device *pdev)
 		}
 
 		if (!mxc_md->link_status) {
-			if (mxc_md->num_sensors > 0) {
+			if (mxc_md->valid_num_sensors > 0) {
 				ret = subdev_notifier_complete(&mxc_md->subdev_notifier);
 				if (ret < 0)
 					goto err_m_ent;
diff --git a/drivers/media/platform/imx8/mxc-media-dev.h b/drivers/media/platform/imx8/mxc-media-dev.h
index 05503fc9aedc..8144b05de67d 100644
--- a/drivers/media/platform/imx8/mxc-media-dev.h
+++ b/drivers/media/platform/imx8/mxc-media-dev.h
@@ -109,6 +109,7 @@ struct mxc_md {
 
 	int  link_status;
 	int  num_sensors;
+	int  valid_num_sensors;
 	u32  nr_isi;
 	bool parallel_csi;
 
-- 
2.17.1

